### PR TITLE
feat: [CDS-52151]: AWS Deploy Step

### PR DESCRIPTION
**Pull request by rohitk-h**
## Describe your changes
- Implement AWS deploy step 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43463)

### CODE REVIEW

**Comment by piyush-bhuwalka** [comment]
Service Spec needs to be moved inside bean directory like all other swimlanes

**Comment by piyush-bhuwalka** [comment]
AwsLambdaManifestVisitorHelper --> should be implemented like GoogleCloudFunctionDefinitionManifestVisitorHelper. 

createDummyVisitableElement function looks different in both

**Review by rohitk-h — COMMENTED** [review]
Agree, it's so that I could test saving of the step with service. We'll rebase based on who merges first to develop. 

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43463#-:-NN_DJ0TEi6wf1wuIkUh:b-fcv8fu)* status: 43 file(s) to review. 0 of 2 approvals obtained.

**Review by PragyeshMishra01 — APPROVED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  public AwsLambdaFunctionsInfraConfig getInfraConfig(InfrastructureOutcome infrastructure, Ambiance ambiance) {
+    return awsLambdaEntityHelper.getInfraConfig(infrastructure, AmbianceUtils.getNgAccess(ambiance));
+  }
+
+  public TaskChainResponse queueTask(StepElementParameters stepElementParameters,
+      AwsLambdaCommandRequest awsLambdaCommandRequest, Ambiance ambiance, PassThroughData passThroughData,
+      boolean isChainEnd) {
+    TaskData taskData = TaskData.builder()
+                            .parameters(new Object[] {awsLambdaCommandRequest})
+                            .taskType(TaskType.AWS_LAMBDA_COMMAND_TASK_NG.name())
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:53]: Lets separate task type for each task. Don't use common task type as it causes issues going forward in terms of backward compatibility.

**Review by rohitk-h — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43463)* status: 44 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 53 at r2](https://reviewable.io/reviews/harness/harness-core/43463#-NNOR8uwOk7oCJGcyJr_-r2-53:-NNcBn-O7FILTUvcBQnT:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/eb9b0619b7dae0dfba68b78a13d395c4140fb126/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L53)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

Lets separate task type for each task. Don't use common task type as it causes issues going forward in terms of backward compatibility.
</blockquote></details>

Done.

**Review by piyush-bhuwalka — APPROVED** [review]
> Reviewed 8 of 24 files at r1, 31 of 35 files at r2, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43463)* status: 45 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java (hunk 1 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java]: This is imp. Lets make sure, each step is independent of each other and have different task types

**Review by sainath-b — APPROVED** [review]
> Reviewed 8 of 24 files at r1, 31 of 35 files at r2, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43463)* status: :shipit: complete! ✓

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 8 of 24 files at r1, 31 of 35 files at r2, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43463)* status: :shipit: complete! ✓

**Pull request merged**